lib/commands/framework-versions/print-versions.ts

```diff
@@ -2,18 +2,19 @@
 "use strict";
 
 export class PrintFrameworkVersionsCommand implements ICommand {
-	constructor(private $cordovaMigrationService: ICordovaMigrationService,
+	constructor(private $cordovaMigrationService: IFrameworkMigrationService,
+		private $nativeScriptMigrationService: IFrameworkMigrationService,
 		private $project: Project.IProject,
 		private $logger: ILogger,
-		private $errors: IErrors) { }
+		private $errors: IErrors,
+		private $projectConstants: Project.IProjectConstants) { }
 
 	public execute(args: string[]): IFuture<void> {
 		return (() => {
-			let supportedVersions: Server.FrameworkVersion[] = this.$cordovaMigrationService.getSupportedFrameworks().wait();
+			let migrationService = this.$project.projectData.Framework === this.$projectConstants.TARGET_FRAMEWORK_IDENTIFIERS.Cordova ? this.$cordovaMigrationService : this.$nativeScriptMigrationService;
```

Comment:
Consider exposing `migrationService` in `project`. This way you can use - `this.$project.migrationService.getSupportedFrameworks`.